default_dial_option_server_option_test.go

```diff
@@ -52,20 +54,31 @@ func (s) TestAddExtraDialOptions(t *testing.T) {
 		cc.Close()
 	}
 
+	// Test the disableGlobalDialOptions dial option. Setting this while dialing
```

Comment:
Can you put this in its own test, please?

Reply:
Sure. I was on the fence about own test but decided to put here to avoid boilerplate. I'll refactor stuff into helper and make this it's own test.